content/docs/0.8.x/reference/cli/_index.md

```diff
@@ -198,3 +198,144 @@ The `KEPTNCONFIG` environment variable holds a config file that contains the cre
 * If `KEPTNCONFIG` environment variable is defined, the Keptn CLI reads the credentials from it. 
 
 * If the credentials are not found in the config file, it will fall back to the credential manager.
+
+## Enabling shell autocompetion of Keptn CLI
+
+Keptn CLI provides autocompletion support for Bash and Zsh, which can save you a lot of typing.
+
+Below are the procedures to set up autocompletion for Bash and Zsh. Please expand the corresponding section matching your configuration.
+
+<details><summary>Autocompletion for Bash (Linux)</summary>
+
+The Keptn CLI completion script for Bash can be generated with the command `keptn completion bash`. Sourcing the completion script in your shell enables Keptn CLI autocompletion.
+
+However, the completion script depends on bash-completion, which means that you have to install this software first (you can test if you have bash-completion already installed by running type _init_completion).
+
+### Install bash-completion for linux
+
+bash-completion is provided by many package managers (see [here](https://github.com/scop/bash-completion#installation)). You can install it with `apt-get install bash-completion` or `yum install bash-completion`, etc.
+
+The above commands create /usr/share/bash-completion/bash_completion, which is the main script of bash-completion. Depending on your package manager, you have to manually source this file in your ~/.bashrc file.
+
+To find out, reload your shell and run type _init_completion. If the command succeeds, you're already set, otherwise add the following to your ~/.bashrc file:
+
+```bash
+source /usr/share/bash-completion/bash_completion
+```
+
+Reload your shell and verify that bash-completion is correctly installed by typing type _init_completion.
+
+### Enable Keptn CLI autocompletion for linux (bash)
+
+You now need to ensure that the Keptn CLI completion script gets sourced in all your shell sessions. There are two ways in which you can do this:
+
+- source the completion script in your ~/.bashrc file:
+
+```bash
+echo 'source <(keptn completion bash)' >>~/.bashrc
+```
+
+- Add the completion script to the /etc/bash_completion.d directory:
+
+```bash
+keptn completion bash >/etc/bash_completion.d/keptn
+```
+
+If you have an alias for Keptn CLI, you can extend shell completion to work with that alias:
+
+```bash
+echo 'alias kc=keptn' >>~/.bashrc
+echo 'complete -F __start_keptn kc' >>~/.bashrc
+```
+
+After reloading your shell, Keptn CLI autocompletion will be enabled succesfully.
+</details>
+
+<details><summary>Autocompletion for Bash (MacOS)</summary>
+
+The Keptn CLI completion script for Bash can be generated with `keptn completion bash`. Sourcing this script in your shell enables Keptn CLI autocompletion.
+
+However, the Keptb CLI autocompletion script depends on bash-completion which you thus have to previously install.
```

Comment:
Typo *Keptb* -> *Keptn*

Reply:
Apologies, missed this while proofreading 😅